x-pack/solutions/search/plugins/search_playground/public/components/summarization_panel/summarization_model.tsx

```diff
@@ -94,6 +95,14 @@ export const SummarizationModel: React.FC<SummarizationModelProps> = ({
       })),
     [models]
   );
+  const currentModel = useMemo(
+    () =>
+      // find the model from the list to ensure all values present vs. what is saved in the form and loaded in local storage.
+      selectedModel !== undefined
+        ? models.find((model) => model.id === selectedModel.id)
+        : undefined,
+    [selectedModel, models]
+  );
```

Comment:
nit:

why not a state var `currentModel` that gets set to `newSelectedModel` in the `onChange` above?